File: DSA/Aanidhay Aggarwal/Day5_q1.java
// Node class for the singly linked list
class Node {
    int data;
    Node next;

    Node(int data) {
        this.data = data;
        this.next = null;
    }
}

class LinkedList {
    Node head;

    // Function to print the elements in reverse order
    void printReverse() {
        printReverse(head);
    }

    // Recursive function to print the elements in reverse order
    void printReverse(Node node) {
        if (node == null) {
            return;
        }
        printReverse(node.next);
        System.out.print(node.data + " ");
    }
}

public class Main {
    public static void main(String[] args) {
        LinkedList list = new LinkedList();
        list.head = new Node(1);
        list.head.next = new Node(2);
        list.head.next.next = new Node(3);

        System.out.print("Linked list: ");
        Node temp = list.head;
        while (temp != null) {
            System.out.print(temp.data + " -> ");
            temp = temp.next;
        }
        System.out.println("null");

        System.out.print("Output: ");
        list.printReverse();
        System.out.println();
    }
}

File: DSA/Hima-Swarupa-Padala/day5q1.py
class Node:
    def __init__(self, data=None):
        self.data = data
        self.next = None

class LinkedList:
    def __init__(self):
        self.head = None

    def append(self, data):
        if not self.head:
            self.head = Node(data)
        else:
            current = self.head
            while current.next:
                current = current.next
            current.next = Node(data)

    def print_reverse(self):
        def _print_reverse(node):
            if node:
                _print_reverse(node.next)
                print(node.data, end=" ")

        _print_reverse(self.head)
        print()  

ll = LinkedList()
ll.append(1)
ll.append(2)
ll.append(3)

print("Linked list:")
current = ll.head
while current:
    print(current.data, end=" ")
    current = current.next
print()  

print("Output:")
ll.print_reverse()